src/Sylius/Component/Core/Repository/ProductRepositoryInterface.php

```diff
@@ -57,4 +57,11 @@ public function findLatestByChannel(ChannelInterface $channel, $locale, $count);
      * @return ProductInterface|null
      */
     public function findOneByChannelAndSlug(ChannelInterface $channel, $locale, $slug);
+
+    /**
+     * @param string $code
+     *
+     * @return ProductInterface|null
+     */
+    public function findOneByCode($code);
```

Comment:
Probably the expresion `expr:service('sylius.repository.product').findOneByCode($productCode)` cannot have as input the array with key `code` 
(see routing in https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/AdminBundle/Resources/config/routing/partial/order.yml).
So yes, we need custom mthod